### PR TITLE
Fix to delegate dtypes.

### PR DESCRIPTION
**Pull request by ueshin**
Fixes to delegate dtypes properly.
The parameter `preserve_dtypes` for the workaround was removed.
Also maked as `TODO` where the dtypes are unknown.

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -2715,7 +2713,9 @@ def add_prefix(self, prefix) -> "Series":
             ]
             + internal.data_spark_columns
         )
-        return first_series(DataFrame(internal.with_new_sdf(sdf)))
+        return first_series(
+            DataFrame(internal.with_new_sdf(sdf, index_dtypes=([None] * internal.index_level)))
```
Comment (xinrong-databricks) [inline databricks/koalas/series.py:2717]: Why don't we pass `object`s as `index_dtypes` directly?

**Review by xinrong-databricks — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
I'm wondering if we wanted to call out `spark_type_to_pandas_dtype` is where the type mapping defined in `internal.py`.
I'm afraid contributors with little context might not be clear about this.

**Comment by xinrong-databricks** [comment]
Looks great! Left some questions. Thank you!

File: databricks/koalas/series.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/series.py:2717]: Why `object`?

If we pass `None`, the dtype will be inferred during the initialization of `InternalFrame`.

**Comment by ueshin** [comment]
Thanks! let me merge this now. Please feel free to leave comments later if any.

**Pull request merged**